test/contracts/sapphire/flashLiquidatorAaveV2.test.ts

```diff
@@ -0,0 +1,149 @@
+
+  before(async () => {
+    // Fork mainnet where vault owner has an open vault
+    await forkNetwork(process.env.MAINNET_ALCHEMY, 15919663);
+    deployer = await impersonateAccount(VAULT_OWNER);
+    liquidator = await impersonateAccount(LIQUIDATOR_ADDRESS);
+
+    // Deploy mock price oracle
+    mockOracle = await new MockSapphireOracleFactory(deployer).deploy();
+
+    // Deploy flash liquidator
+    liquidatorContract = await new FlashLiquidatorAaveV2Factory(
+      deployer,
+    ).deploy(
+      '0xB53C1a33016B2DC2fF3653530bfF1848a515c8c5',
+      '0xe592427a0aece92de3edee1f18e0157c05861564',
+      LIQUIDATOR_ADDRESS,
+    );
+
+    // Turn on reporting of real timestamp
+    await mockOracle.setReportRealTimestamp(true);
+  });
+
+  it('executes liquidation', async () => {
+    let coreAsAdmin = SapphireCoreV1Factory.connect(CORE_ADDRESS, deployer);
+    const adminAddress = await coreAsAdmin.getAdmin();
+    coreAsAdmin = SapphireCoreV1Factory.connect(
+      CORE_ADDRESS,
+      await impersonateAccount(adminAddress),
+    );
+
+    // Confirm the vault is safe
+    const oracle = ISapphireOracleFactory.connect(
+      await coreAsAdmin.oracle(),
+      liquidator,
+    );
+    const currentPrice = await oracle.fetchCurrentPrice();
+    const assessor = SapphireAssessorFactory.connect(
+      await coreAsAdmin.assessor(),
+      liquidator,
+    );
+    const assessedCRatio = await assessor.assess(
+      await coreAsAdmin.lowCollateralRatio(),
+      await coreAsAdmin.highCollateralRatio(),
+      VAULT_OWNER_CREDIT_SCORE_PROOF,
+      true,
+    );
+    expect(
+      await coreAsAdmin.isCollateralized(
+        VAULT_OWNER,
+        currentPrice.price,
+        assessedCRatio,
+      ),
+    ).to.be.true;
+
+    // Set the mock oracle
+    await setAccountBalance(adminAddress, utils.parseEther('1'));
+    await coreAsAdmin.setOracle(mockOracle.address);
+    // Set a low price. The current liquidation price is $963
+    const newPrice = utils.parseEther('960');
+    await mockOracle.setPrice(newPrice);
+
+    // Confirm the vault is unsafe
+    expect(
+      await coreAsAdmin.isCollateralized(VAULT_OWNER, newPrice, assessedCRatio),
+    ).to.be.false;
+
+    // Capture the usdc balance before liquidation
+    const usdc = IERC20Factory.connect(USDC_ADDRESS, liquidator);
+    const profitReceiverAddress = await liquidatorContract.profitReceiver();
+    const initialUsdcBalance = await usdc.balanceOf(profitReceiverAddress);
+
+    await liquidatorContract.liquidate(
+      CORE_ADDRESS,
+      VAULT_OWNER,
+      VAULT_OWNER_CREDIT_SCORE_PROOF,
+    );
+
+    // Confirm the usdc balance increased
+    const finalUsdcBalance = await usdc.balanceOf(profitReceiverAddress);
+    expect(finalUsdcBalance).to.be.gt(initialUsdcBalance);
+    // console.log(
+    //   `I've made ${utils.formatUnits(
+    //     finalUsdcBalance.sub(initialUsdcBalance),
+    //     6,
+    //   )} USDC`,
+    // ); // $3.76
```

Comment:
🎯

Reply:
Left there intentionally in case we want to debug later.

Reply:
We have source-control for that reason. We don't leave comments in code for later debugging in any other repo, is there any particular reason why it should be acceptable in contracts repo?